backend/task/data_from_fire.py

```diff
@@ -0,0 +1,188 @@
+        logger.info(f"To be crawled urls:{[str(fire_event) for fire_event in to_crawl]}")
+        # clean up the temporary directory, in case of a dirty directory
+        self.crawler.cleanup()
+        # records with errors are stored in errors list for a backup
+        errors = []
+        # get the latest fire_id, if there is no fire id exists, then fire_id = 0
+        fire_id = 0 if self.dumper.get_latest_fire_id() is None else int(self.dumper.get_latest_fire_id() + 1)
+        logger.info(f"Initial fire id: {fire_id}")
+        # entries in to_crawl are:
+        # New tuples: new tuples that is never crawled: (year, state, urlname)
+        # Recent tuples: tuples of recent fires: (old_id, year, state, urlname)
+        for fire_event in to_crawl:
+            # if statement take forward here, handle 3 or 4 values
+            logger.info(f"Now working on: {fire_event}")
+            fire_id, year, state, urlname = fire_event.to_tuple(fire_id)
+            # generate the url from known information
+            # download all records to temporary directory
+            self.crawler.crawl(fire_event.to_url())
+            # if the temporary directory has more than one sub directory(records), then the fire is a sequence of fire
+            if_sequence = False if len([f for f in os.listdir(FIRE_DATA_DIR) if not f.startswith('.')]) == 1 else True
+            # for each record in the temporary directory:
+            for record in [f for f in os.listdir(FIRE_DATA_DIR) if not f.startswith('.')]: # glob
+                absolute_path_folder = os.path.join(FIRE_DATA_DIR, record)
+                # extract the shapefile into a dictionary called single_record
+                try:
+                    single_record = self.extractor.extract(absolute_path_folder, if_sequence, fire_id, state)
+                except IncompleteShapefileError as err:
+                    # the extractor result can be empty when the record is incomplete and cannot be decoded
+                    # in this situation, skip the record
+                    logger.warning(str(err))
+                    continue
+                # insert the single record into fire table
+                self.dumper.insert(single_record)
+                logger.info(f"Successfully inserted {record} into fire_info.")
+            # after the for loop, all records belongs to this fire should be inserted into fire
+            # then we can create the merged record and insert it into fire_merged and mark it as crawled in fire_history
+            fire_id = self.get_fire_id(fire_event,
+                      self.merge_fire_and_return_fire_id(fire_id, year, urlname, state, current_year, errors))
+            logger.info(f"New fire id : {fire_id}")
+            # clean up the temporary directory for crawling the next fire record
+            self.crawler.cleanup()
+        # finished all insertion
+        logger.info("Finished running. Waiting for one day.")
```

Comment:
no need to log out "waiting" time. just log it is finished. TaskManager will take care of repeating.